modules/key-card/src/generateQrData.ts

```diff
@@ -0,0 +1,171 @@
+import { BaseCoin } from '@bitgo/statics';
+import { Keychain } from '@bitgo/sdk-core';
+import { encrypt } from '@bitgo/sdk-api';
+import * as assert from 'assert';
+
+export interface GenerateQrDataParams {
+  // The backup keychain as it is returned from the BitGo API upon creation
+  backupKeychain: Keychain,
+  // The 3rd party provider of the backup key if neither the user nor BitGo stores it
+  backupKeyProvider?: string;
+  // The key id of the backup key, only used for cold keys
+  backupMasterKey?: string;
+  // The BitGo keychain as it is returned from the BitGo API upon creation
+  bitgoKeychain: Keychain,
+  // The coin of the wallet that was/ is about to be created
+  coin: Readonly<BaseCoin>;
+  // A code that can be used to encrypt the wallet password to.
+  // If both the passphrase and passcodeEncryptionCode are passed, then this code encrypts the passphrase with the
+  // passcodeEncryptionCode and puts the result into Box D. Allows recoveries of the wallet password.
+  passcodeEncryptionCode?: string;
+  // The wallet password
+  // If both the passphrase and passcodeEncryptionCode are passed, then this code encrypts the passphrase with the
+  // passcodeEncryptionCode and puts the result into Box D. Allows recoveries of the wallet password.
+  passphrase?: string;
+  // The user keychain as it is returned from the BitGo API upon creation
+  userKeychain: Keychain;
+  // The key id of the user key, only used for cold keys
+  userMasterKey?: string;
+}
+
+interface QrDataEntry {
+  data: string;
+  description: string;
+  image: string;
+  title: string;
+  publicMasterKey?: string;
+}
+
+export interface QrData {
+  backup?: QrDataEntry;
+  bitgo?: QrDataEntry;
+  passcode?: QrDataEntry;
+  user: QrDataEntry;
+}
+
+function getPubFromKey(key: Keychain): string | undefined {
+  if (key.type === 'blsdkg' && key.commonPub) {
+    return key.commonPub;
+  }
+  if (key.type === 'tss' && key.commonKeychain) {
+    return key.commonKeychain;
+  }
+  if (key.type === 'independent' && key.pub) {
+    return key.pub;
+  }
```

Comment:
```suggestion
  switch(key.type) {
      case 'blsdkg':
         return key.commonPub;
       ....
  }
```
nit: the check to see if `commonPub` and other fields exists is redundant since this would return `undefined` for those cases anyways.

Reply:
Great idea, will do